VxWorks/Os/ConditionVariable.cpp

```diff
@@ -0,0 +1,57 @@
+// ======================================================================
+// \title VxWorks/Os/ConditionVariable.cpp
+// \brief VxWorks implementations for Os::ConditionVariable
+// ======================================================================
+#include "VxWorks/Os/ConditionVariable.hpp"
+#include "Fw/Types/Assert.hpp"
+#include "VxWorks/Os/Mutex.hpp"
+#include "VxWorks/Os/error.hpp"
+
+namespace Os {
+namespace VxWorks {
+namespace Mutex {
+
+VxWorksConditionVariable::VxWorksConditionVariable() {
+    this->m_handle.m_condition = condVarCreate(CONDVAR_Q_PRIORITY);
```

Comment:
This `condVarCreate` call is designed for userspace. As such, it has an implicit `malloc` of the memory underneath.  We should either:

1. Document this in the README/SDD/and/or HPP comments
2. Switch to a static condition var object and use `condVarInit` to inialize the stack allocated memory

Reply:
Given these variables themselves are stack allocated, it is probably not a big deal to choose either case.  @timcanham do you have a preference here? @garthwatney ?

Reply:
Since most OSAL calls allocate OS/kernel resources, documentation is likely the best path.

Reply:
I'll capture this in the header file as a doxygen style comment